src/main/webapp/query-control.js

```diff
@@ -35,13 +35,82 @@ function clearPreviousResult() {
   document.getElementById('result').style.display = 'block';
 }
 
-function getFetchUrl(personType, action, location, year) {
-  return '/query?person-type=' + personType +
+function getFetchUrl(servlet, personType, action, location, year) {
+  return '/' + servlet + '?person-type=' + personType +
     '&action=' + action +
     '&location=' + location +
     '&year=' + year;
 }
 
+function getTitle(personType, location, year, locationInput, actionInput) {
+  const isCountyQuery = location !== 'state';
+  const region = isCountyQuery ? locationInput + ' county' : 'U.S. state';
+  const title = 'Population who ' + actionInput +
+    ' each ' + region + ' (' +
+    personType.replace('-', ' ') + ')' +
+    ' in ' + year;
+  return title;
+}
+
+// getHistory fetches to the History Servlet and returns a list of 
+// links of queries that the logged in user has visited before
+function getHistory(personType, action, location, year, userId) {
+  let fetchUrl = getFetchUrl('history', personType, action, location, year);
+  fetchUrl = fetchUrl + '&user-id=' + userId;
+  
+  // clear previous results
+  const historyContainer = document.getElementById('history');
+  historyContainer.innerHTML = '';
+  const header = document.createElement('p');
+  header.innerText = "Pages you've Viewed";
+  historyContainer.appendChild(header);
+
+  // change back the action to be gramatically correct in the title
+  const gramaticallyCorrectAction = new Map();
+  gramaticallyCorrectAction.set(
+        'live', 'lived in',
+      ).set(
+        'work', 'worked in',
+      ).set(
+        'moved', 'moved to',
+      );
+  fetch(fetchUrl).then((response) => response.json()).then(
+    function (jsonResponse) {
+      jsonResponse.forEach((historyElement) => {
+        if (historyElement !== null) {
+          let historyTextNode;
+          if (historyElement.location !== 'state') {
```

Comment:
You could shorten this/remove some repetition by doing something like
location = historyElement.location === 'state' ? 'Each U.S. state' : stateInfo[historyElement.location].name
and then call getTitle with that location variable

Reply:
good idea! done